# Trabajo Práctico 1 - Matrices de Insumo Producto

### Consigna 1 - 

Analizar qué sucede con la solución del sistema 2 en cada uno de los siguientes casos:

Sistema 2:

$$p=(I − A)^{-1}d = Ld$$

a) $(I − A)$ es inversible :

$\quad$ Si la matriz $(I − A)$ es inversible, el sistema $(I-A)^{-1}$ tiene una única solución. Esto, en términos de lo que el sistema representa, significa que existe una única producción $\textbf{p}$ que satisface tanto la demanda interna como la externa.

b) $(I − A)$  es inversible y $\textbf{d}$ = 0 :

$\quad$ Que $\textbf{d}$ sea 0 significa que estamos hablando de una economía cerrada. Por lo tanto, toda la producción esta de destinada a satisfacer la demanda interna. Además, la matriz $\\$ $(I-A)$ es inversible, lo que implica que podemos encontrar la producción de equilibrio para afrontar la demanda interna. $\\$ $\quad$ Otro enfoque, que podriamos darle es que el vector $\textbf{d}$ que representa la demanda externa, la demanda no destinada a la producción de otros bienes, sea 0. Es decir que dentro del país no se precisa ninguno de los bienes. Por lo tanto, tiene sentido que el vector de producción sea 0, ya que no se produce lo que no se demanda.


c) $(I − A)$ no es inversible :

 $\quad$ Analizando unicamente el sistema 2, en este caso se deduce que no existe una solucion $\textbf{p}$ que satisface tanto la demanda interna como la externa.

 $\quad$ Sin embargo, si vemos el sistema 1, podemos llegar a la ecuacion:

$$
(I - A) \cdot p = d \quad \text{(1)}
$$

 $\quad$ Luego, como suponemos que $(I − A)$ no es inversible obtenemos 2 posibles alternativas: que tenga infinitas soluciones o que no tenga solucion. Por lo que puede llegar a pasar que hayan infinitas producciones que satisfacen tanto la demanda externa como interna o puede no haber ninguna.


d) $(I − A)$ no es inversible y $\textbf{d}$ = 0 :

$\quad$ Este caso es similar a dos de los anteriores. Por un lado, el vector $\textbf{d}$ es 0, que como ya mencionamos representa la demanda que no busca producir más bienes, y esto indicaba una falta demanda de producto impedia la producción de los mismos. Por otro lado, la no inversibilidad de la matriz de Leontief nos dice que aunque los productos fueran demandados no existiria una única producción de equilibrio, pudiendo existir infinitas combinaciónes que permitieran alcanzarlo. La opción de no existir queda descartada en este caso ya que no producír nada ya es una solución para esta economía
    

### Consigna 3 -  
Resolver el sistema 2 utilizando inversaLU para los siguientes valores de $\textbf{A}$ y $\textbf{d}$:

$$
\mathbf{A} = 
\begin{pmatrix}
0.3 & 0.0 & 0.1 \\
0.05 & 1.0 & 0.2 \\
0.1 & 0.15 & 0.1
\end{pmatrix},
\quad
\mathbf{d} =
\begin{pmatrix}
100 \\
100 \\
300
\end{pmatrix}
$$


1) ¿Hay una justificación económica de los valores encontrados para $\textbf{p}$?
2) ¿Cual es la característica de la matriz que lleva a esta solución?

In [13]:
from funciones import inversaLU, calcularLU
import numpy as np

A = np.array([[0.3, 0.0, 0.1], [0.05, 1.0, 0.2], [0.1, 0.15, 0.1]])

d = [[100], [100], [300]]

I = np.eye(3)

L = inversaLU(I-A)

p = L@d

print(p)

[[   68.96551724]
 [-5149.42528736]
 [ -517.24137931]]


1)  $\quad$ Se ve claramente que los sectores que dependen en alguna medida del producto 2 presentan valores extraños. Esto se debe que el producto 2 se requiere en 1.15 unidades entre los sectores 2 y 3. Mientras que solo se puede generar en una unidad. $\\$

$\quad$ Por un lado, la producción de producto 2 nos da un valor de alrededor de 5000 en modulo porque es una demanda que no esta siendo satisfecha. Por otro lado, volvamos a que la demanda total de producto 2 es de 1.15 unidades sumando la demanda de los sectores 2 y 3, pero recordemos que la producción de producto 2 necesita una unidad de si mismo. Entonces si restamos la producción de producto 2 y su demanda obtenemos un resultado con signo negativo. Esto es indicador de que nunca vamos a poder producir lo necesario para satisfacer la demanda total para este producto. $\\$ $\quad$ En resumen, las desviaciones observadas en los sectores 2 y 3 son producto de demandar en cantidades que no pueden ser satisfechas al producto 2. Esto derivó en una economía inviable. Los altos valores en modulo indican la necesidad producir más para intentar satisfacer su demanda de producto 2 y el simbolo menos nos indica que a pesar de producir lo máximo nunca se llego a satisfacer esa demanda.

2)  Para calcular el determinante de una matriz 3×3, se puede usar la siguiente fórmula general: $\\$ Para una matriz de la forma:
    $$
    \mathbf{A} = 
    \begin{pmatrix}
    a & b & c \\
    d & e & f \\
    g & h & i
    \end{pmatrix}
    $$

    El determinante se calcula como: det=a(ei−fh)−b(di−fg)+c(dh−eg)

    En este caso. la matriz es:
    $$
    \mathbf{A} = 
    \begin{pmatrix}
    1.37931034 & -0.68965517 & 0 \\
    -2.98850575 & -28.50574713 & -6.66666667 \\
    -0.34482759 & -4.82758621 & 0
    \end{pmatrix}
    $$

    Sustituyendo la fórmula: $\\$
    det= 1.37931034⋅(−32.1848725)−((−0.68965517)⋅(−2.2955691)) = −46.0115


### Consigna 4 - 

Analizar analíticamente qué sucede con la producción total cuando a partir de una demanda externa d, la demanda externa para el producto 3 cambia en un ∆d. La demanda externa se convierte en
$$d'=d+\begin{pmatrix}
    0 \\
    0 \\
    1
    \end{pmatrix}$$

Obtener la expresión del vector ∆p correspondiente a este ejemplo.

Conocemos la economía con $\textbf{d}$, que esta modelada por:
$$\mathbf{p}= \mathbf{Ld}$$
Ahora calculemos la economía con $\textbf{d'}$

$$\mathbf{p'} = \mathbf{Ld'} = \mathbf{L(d+∆d)}$$

$$\mathbf{p + ∆p} = \mathbf{Ld} + \mathbf{L∆d}$$

Sabiendo que $\textbf{p = Ld}$ cancelamos y obtenemos que:

$$\mathbf{∆p} = \mathbf{L∆d} \\$$
Reemplazando :
$$
\mathbf{∆d} = \begin{pmatrix}
    0 \\
    0 \\
    1
    \end{pmatrix}
$$

y obtenemos

$$\mathbf{∆p} = \mathbf{L} \begin{pmatrix}
    0 \\
    0 \\
    1
    \end{pmatrix}\\$$

Lo cual implica un aumento en la demanda del producto 3.


### Consigna 5 - 

Encontrar los coeficientes técnicos A y la matriz de Leontief para la economía dada.

En primer lugar, escribimos la matriz Z 

$$
    \mathbf{Z} = 
    \begin{pmatrix}
    350 & 0 & 0 \\
    50 & 250 & 150 \\
    200 & 150 & 550
    \end{pmatrix}
$$

Luego, podemos escribir la matriz P como 

$$
    \mathbf{P} = 
    \begin{pmatrix}
    1000 & 0 & 0 \\
    0 & 500 & 0 \\
    0 & 0 & 1000
    \end{pmatrix}
$$

Entonces, por lo visto, $$\mathbf{A} = ZP^{-1}$$

Calculamos A:




In [14]:
Z=np.array([[250,0,0],[50,250,150],[200,150,550]])
P=np.array([[1000,0,0],[0,500,0],[0,0,1000]])
A=Z@inversaLU(P)

print(A)

[[0.25 0.   0.  ]
 [0.05 0.5  0.15]
 [0.2  0.3  0.55]]


In [15]:
I = np.eye(3)

L = inversaLU(I-A)

print(L)

[[1.33333333 0.         0.        ]
 [0.38888889 2.5        0.83333333]
 [0.85185185 1.66666667 2.77777778]]


Entonces, la matriz de Leontief queda de la siguiente manera:
$$
    \mathbf{L} = (I-A)^{-1} = 
    \begin{pmatrix}
    1.53846154 & 0 & 0 \\
    0.8974359 & 2.5 & 1.66666667 \\
    0.98290598 & 0.83333333 & 2.77777778
    \end{pmatrix}
$$


### Consigna 6 -
A partir de la fórmula 4, deducir cómo se calcula la variación de la producción en la región $\textit{r} $ para cubrir una variación de la demanda considerando las relaciones interregionales. $\\$ Fijando por simplicidad $\mathbf{∆d^{s}}$ = 0, deducir la siguiente fórmula:
$$
\mathbf{∆p^{r}} = (I - \mathbf{A^{rr}} - \mathbf{A^{rs}}(I - \mathbf{A^{ss}})^{-1}\mathbf{A^{sr}})^{-1}\mathbf{∆d^{r}}
$$

En primer lugar, queremos obtener los valores originales de $\mathbf{p^{r}}$ y $\mathbf{p^{s}}$ en funcion de $\mathbf{d^{r}}$ y $\mathbf{p^{s}}$. $\\$ 
Luego, analizamos que ocurre con $\mathbf{p^{r'}}$ = $\mathbf{p^{r}}$ + $\mathbf{∆p^{r}}$ para $\mathbf{d^{r'}}$ = $\mathbf{d^{r}}$ + $\mathbf{∆d^{r}}$


$$
    \begin{pmatrix}  
    \begin{pmatrix}
    I & 0 \\
    0 & I \\
    \end{pmatrix} - 
    \begin{pmatrix}
    \mathbf{A^{rr}} & \mathbf{A^{rs}} \\
    \mathbf{A^{sr}} & \mathbf{A^{ss}} \\
    \end{pmatrix} 
    \end{pmatrix} \cdot
    \begin{pmatrix}
    \mathbf{p^{r}}\\
    \mathbf{p^{s}} \\
    \end{pmatrix} 
    = 
    \begin{pmatrix}
    \mathbf{d^{r}}\\
    \mathbf{d^{s}} \\
    \end{pmatrix} 
$$

Luego,
- ($\textbf{I}$- $\mathbf{A^{rr}}$)$\mathbf{p^{r}}$ - $\mathbf{A^{rs}}$ $\mathbf{p^{s}}$ = $\mathbf{d^{r}}$
- (-$\mathbf{A^{sr}}$ $\mathbf{p^{r}}$) + ($\textbf{I}$-$\mathbf{A^{ss}}$)$\mathbf{p^{s}}$ = $\mathbf{d^{s}}$

$\implies$ $\mathbf{p^{s}}$ = $(\textbf{I} -\mathbf{A^{ss}})^{-1}$($\mathbf{d^{s}}$ + $\mathbf{A^{sr}}$ $\mathbf{p^{r}}$)  $\\$ 
$\implies$ $\mathbf{d^{r}}$ = ($\textbf{I}$- $\mathbf{A^{rr}}$)$\mathbf{p^{r}}$  $\mathbf{-A^{rs}}$ $(\textbf{I} -\mathbf{A^{ss}})^{-1}$($\mathbf{d^{s}}$ + $\mathbf{A^{sr}}$ $\mathbf{p^{r}}$) $\quad$ $\textbf{(1)}$

Luego con $\textbf{∆d}$ $\\$ 
($\textbf{I}$- $\mathbf{A^{rr}}$)($\mathbf{p^{r}}$ + $\mathbf{∆p^{r}}$) - $\mathbf{A^{rs}}$ $\mathbf{p^{s}}$ = $\mathbf{d^{r}}$ + $\mathbf{∆d^{r}}$ $\\$ 
-$\mathbf{A^{sr}}$($\mathbf{p^{r}}$ + $\mathbf{∆p^{r}}$) + ($\textbf{I}$-$\mathbf{A^{ss}}$)$\mathbf{p^{s}}$ = $\mathbf{d^{s}}$ $\implies$
$\mathbf{p^{s}}$  = $(\textbf{I} -\mathbf{A^{ss}})^{-1}$($\mathbf{d^{s}}$ + $\mathbf{A^{sr}}$ $\mathbf{p^{r}}$ + $\mathbf{A^{sr}}$ $\mathbf{∆p^{r}}$)

($\textbf{I}$- $\mathbf{A^{rr}}$)($\mathbf{p^{r}}$ + $\mathbf{∆p^{r}}$)$\mathbf{-A^{rs}}$ $(\textbf{I} -\mathbf{A^{ss}})^{-1}$($\mathbf{d^{s}}$ + $\mathbf{A^{sr}}$ $\mathbf{p^{r}}$ + $\mathbf{A^{sr}}$ $\mathbf{∆p^{r}}$) = $\mathbf{d^{r}}$ + $\mathbf{∆d^{r}}$ $\\$ 

($\textbf{I}$- $\mathbf{A^{rr}}$)$\mathbf{p^{r}}$ + ($\textbf{I}$- $\mathbf{A^{rr}}$)$\mathbf{∆p^{r}}$ $\mathbf{-A^{rs}}$ $(\textbf{I} -\mathbf{A^{ss}})^{-1}$ $\mathbf{d^{s}}$ $\mathbf{-A^{rs}}$ $(\textbf{I} -\mathbf{A^{ss}})^{-1}$ $\mathbf{A^{sr}}$ $\mathbf{p^{r}}$ $\mathbf{-A^{rs}}$ $(\textbf{I} -\mathbf{A^{ss}})^{-1}$ $\mathbf{A^{sr}}$ $\mathbf{∆p^{r}}$ = $\mathbf{d^{r}}$ + $\mathbf{∆d^{r}}$ $\\$ 

Por $\textbf{(1)}$, $\mathbf{d^{r}}$ = ($\textbf{I}$- $\mathbf{A^{rr}}$)$\mathbf{p^{r}}$ $\mathbf{-A^{rs}}$ $(\textbf{I} -\mathbf{A^{ss}})^{-1}$ $\mathbf{d^{s}}$ $\mathbf{-A^{rs}}$ $(\textbf{I} -\mathbf{A^{ss}})^{-1}$ $\mathbf{A^{sr}}$ $\mathbf{p^{r}}$

$\implies$ $\mathbf{∆d^{r}}$ = ($\textbf{I}$- $\mathbf{A^{rr}}$)$\mathbf{∆p^{r}}$ $\mathbf{-A^{rs}}$ $(\textbf{I} -\mathbf{A^{ss}})^{-1}$ $\mathbf{A^{sr}}$ $\mathbf{∆p^{r}}$ $\\$ 
$\implies$ $\mathbf{∆d^{r}}$ =  ($\textbf{I}$- $\mathbf{A^{rr}}$ $\mathbf{-A^{rs}}$ $(\textbf{I} -\mathbf{A^{ss}})^{-1}$ $\mathbf{A^{sr}}$)$\mathbf{∆p^{r}}$ $\\$ 
$\implies$ $\mathbf{∆p^{r}}$ = $(\textbf{I}- \mathbf{A^{rr}} \mathbf{-A^{rs}} (\textbf{I} -\mathbf{A^{ss}})^{-1} \mathbf{A^{sr}})^{-1}$ $\mathbf{∆d^{r}}$

### Consigna 7 -

- Seleccionar los dos países $\textit{P1}$ (Venezuela) y $\textit{P2}$ (Costa Rica) que les fue asignado al grupo y generar la matriz de
Insumo-Producto de estas dos regiones.

Importamos las librerias


In [20]:
import pandas as pd
from inline_sql import sql, sql_val
import numpy as np
from funciones import inversaLU
import matplotlib.pyplot as plt

Leemos el excel y separamos nuestros paises

In [ ]:
datos_basicos = pd.read_excel("matrizlatina2011_compressed_0.xlsx", sheet_name= "LAC_IOT_2011")


#Separamos los datos de las filas de nuestros 2 paises
datos_impotantes_rel= sql^ """
                        SELECT *
                        FROM datos_basicos
                        WHERE Country_iso3 LIKE 'CRI%' or Country_iso3 LIKE 'VEN%' 
                    """ 
                    
# Contruimos una tabla auxiliar con todos los datos
#Creamos un dataframe con los paises y los sectores
datos_relacion_iso = pd.concat([datos_impotantes_rel['Country_iso3'], datos_impotantes_rel['Nosector']], axis = 1)              
#filtramos los datos de las columnas de relacion con Venezuela
datos_relacion_filtrados = pd.concat([datos_relacion_iso, datos_impotantes_rel.filter(regex="^VEN").copy()], axis = 1)
#filtramos los datos de las columnas de relacion con Costa Rica
datos_relacion= pd.concat([datos_relacion_filtrados, datos_impotantes_rel.filter(regex="^CRI").copy()], axis = 1)
#agregamos los datos del output
datos_relacion = pd.concat([datos_relacion, datos_impotantes_rel['Output']], axis = 1)

Solucion a los $output=0$

In [ ]:
#las columnas que involucran a ambos paises
columnas = pd.concat([datos_relacion.filter(regex="^CRI"), datos_relacion.filter(regex="^VEN")], axis = 1).columns
# Procesamos los datos un poco porque nos encontramos con la dificultad de que los output valian 0 en esas columnas.
# Pero los valores en dichas filas no eran exclusivamente 0. Entonces los igualamos a 0, ya que eran del orden de 10^(-5)

datos_relacion.loc[(datos_relacion['Country_iso3'] == 'VEN') & (datos_relacion['Nosector'] == 's28'), columnas] = 0

datos_relacion.loc[(datos_relacion['Country_iso3'] == 'VEN') & (datos_relacion['Nosector'] == 's31'), columnas] = 0

datos_relacion.loc[(datos_relacion['Country_iso3'] == 'CRI') & (datos_relacion['Nosector'] == 's31'), columnas] = 0

datos_relacion.loc[(datos_relacion['Country_iso3'] == 'CRI') & (datos_relacion['Nosector'] == 's03'), columnas] = 0


Separamos los datos en 4 dataframes

In [ ]:
# Construimos un df con Todos los valores de todos los sectores de ambos paises
numeros_relacion = pd.concat([datos_relacion.filter(regex="^CRI"),  datos_relacion.filter(regex="^VEN")], axis = 1)
#Separamos las filas solo de  Costa Rica
numeros_CRI = sql^ """
                        SELECT *
                        FROM datos_relacion
                        WHERE Country_iso3 LIKE 'CRI%'
                    """ 
#Datos de Costa Rica vs Costa Rica
numeros_internos_CRI = numeros_CRI.filter(regex="^CRI")
#Datos de Costa Rica vs Venezuela
numeros_externos_CRI = numeros_CRI.filter(regex="^VEN")
#Analogamente con los de Venezuela
numeros_VEN = sql^ """
                        SELECT *
                        FROM datos_relacion
                        WHERE Country_iso3 LIKE 'VEN%'
                    """ 

numeros_internos_VEN = numeros_VEN.filter(regex="^VEN")

numeros_externos_VEN = numeros_VEN.filter(regex="^CRI")

Pasamos de DataFrame a np.array

In [ ]:
Z_CRICRI = numeros_internos_CRI.to_numpy()

Z_VENVEN = numeros_internos_VEN.to_numpy()


Z_CRIVEN = numeros_externos_CRI.to_numpy()

Z_VENCRI = numeros_externos_VEN.to_numpy()


matirz_Z_grande = numeros_relacion.to_numpy()

- Calcular los coeficientes técnicos para cada bloque en sus submatrices intra-regionales
e inter-regionales de la A total

Conseguimos las Matrices P

In [ ]:
P_VEN=np.zeros([40,40])
#Tomamos las columnas Output
Output_VEN = sql^ """
                        SELECT Output
                        FROM datos_relacion
                        WHERE Country_iso3 LIKE 'VEN%'
                    """
                    
Output_VEN = Output_VEN.to_numpy()
#Construimos P para Venezuela
for fila in range (P_VEN.shape[0]):
    if Output_VEN[fila]==0:
        P_VEN[fila][fila]=1
    else:
        P_VEN[fila][fila]= Output_VEN[fila][0]

#De la misma manera para Costa Rica
P_CRI=np.zeros([40,40])

Output_CRI= sql^ """
                        SELECT Output
                        FROM datos_relacion
                        WHERE Country_iso3 LIKE 'CRI%'
                    """
                    
Output_CRI = Output_CRI.to_numpy()
for fila in range (P_CRI.shape[0]):
    if Output_CRI[fila]==0:
        P_CRI[fila][fila]=1
    else:
        P_CRI[fila][fila]=Output_CRI[fila][0]

In [ ]:
#Tomamos las inversas
P_INV_CRI=inversaLU(P_CRI)
P_INV_VEN=inversaLU(P_VEN)

# Ejercicio 7 Matrices Insumo-Producto
A_VV= Z_VENVEN@P_INV_VEN

A_VC = Z_VENCRI@P_INV_CRI

A_CC= Z_CRICRI@P_INV_CRI

A_CV = Z_CRIVEN@P_INV_VEN

Completamos la A del sistema completo

In [ ]:
A_grande = np.zeros([80,80])

for fila in range(80):
    for columna in range(80):
        if (fila < 40) and (columna < 40):
            A_grande[fila][columna] = A_VV[fila][columna]
        elif (fila < 40):
            A_grande[fila][columna] = A_VC[fila][columna -40]
        elif (fila >= 40) and (columna < 40):
            A_grande[fila][columna] = A_CV[fila -40][columna]
        else:
            A_grande[fila][columna] = A_CC[fila -40][columna -40]

- Simular un shock negativo sobre el sector s05 del 10 % del total del sector, y un shock positivo del 3.3 % del total de cada uno de los siguientes sectores: s06, s07 y s08. Todos estos shocks se producen en el país P1

Simulamos el shock para el modelo simple

In [ ]:
I = np.eye(40)
#Calculamos la demanda
d_VEN_simple = (I- A_VV)@Output_VEN
#Calulcamos el shock en la demanda
d_VEN_shock_simple = d_VEN_simple.copy()

d_VEN_shock_simple[4] = d_VEN_shock_simple[4]*0.9
d_VEN_shock_simple[5] = d_VEN_shock_simple[5]*1.033
d_VEN_shock_simple[6] = d_VEN_shock_simple[6]*1.033
d_VEN_shock_simple[7] = d_VEN_shock_simple[7]*1.033


#Calculo del delta
delta_d_simple = d_VEN_shock_simple - d_VEN_simple
#Calculamos la producción segun lo desarollado en ecuación 5
delta_p_simple= inversaLU(I - A_VV) @ delta_d_simple

p_prima_simple = Output_VEN + delta_p_simple

Simulamos el shock para el modelo completo

In [ ]:
#%% Simulamos el shock para el modelo completo

I = np.eye(40)
#Calculamos la demanda de venezuela
d_VEN_completo = (I- A_VV)@Output_VEN
#Calculamos el shock
d_VEN_shock_completo = d_VEN_completo.copy()

d_VEN_shock_completo[4] = d_VEN_shock_completo[4]*0.9
d_VEN_shock_completo[5] = d_VEN_shock_completo[5]*1.033
d_VEN_shock_completo[6] = d_VEN_shock_completo[6]*1.033
d_VEN_shock_completo[7] = d_VEN_shock_completo[7]*1.033

#sacamos el delta
delta_d_completo = d_VEN_shock_completo - d_VEN_completo
#usamos lo deducido en la consigna 6 ya que la demanda Costa Rica no se modifica
delta_p_completo = inversaLU(I- A_VV - (A_VC @ inversaLU(I - A_CC) @ A_CV)) @ delta_d_completo

p_prima_completo = Output_VEN + delta_p_completo

- Analizar cómo se modifica la producción cuando se considera el modelo de región simple
(ecuación 5) y cuando se considera la fórmula completa de las 2 regiones (ecuación 6).

Realizamos un grafico de barras comparando los resultados para el modelo simple

In [ ]:
x = np.arange(1, 41)

# Definir el ancho de las barras
width = 0.35

# Crear el gráfico de barras
fig, ax = plt.subplots(figsize=(12, 6))

# Dibujar las barras del primer array
bars1 = ax.bar(x - width/2, np.ravel(Output_VEN), width, label='p_simple')

# Dibujar las barras del segundo array
bars2 = ax.bar(x + width/2, np.ravel(p_prima_simple), width, label='p_prima_simple')

ax.set_xlabel('Fila')
ax.set_ylabel('Valores')
ax.set_title('Comparación de Arrays')
ax.set_xticks(x)
ax.set_xticklabels([f'{i}' for i in x])

# Añadir leyenda
ax.legend()

# Mostrar el gráfico
plt.show()

Realizamos un grafico de barras comparando los resultados para el modelo completo

In [ ]:
x = np.arange(1, 41)

# Definir el ancho de las barras
width = 0.35

# Crear el gráfico de barras
fig, ax = plt.subplots(figsize =(12 , 6))

# Dibujar las barras del primer array
bars1 = ax.bar(x - width/2, np.ravel(Output_VEN), width, label='p_completo')

# Dibujar las barras del segundo array
bars2 = ax.bar(x + width/2, np.ravel(p_prima_completo), width, label='p_prima_completo')

ax.set_xlabel('Fila')
ax.set_ylabel('Valores')
ax.set_title('Comparación de Arrays')
ax.set_xticks(x)
ax.set_xticklabels([f'{i}' for i in x])

# Añadir leyenda
ax.legend()

# Mostrar el gráfico
plt.show()


Y comparando ambas variaciones entre si 

In [ ]:
x = np.arange(1, 41)

# Definir el ancho de las barras
width = 0.35

# Crear el gráfico de barras
fig, ax = plt.subplots(figsize=(12 ,6))

# Dibujar las barras del primer array
bars1 = ax.bar(x - width/2, np.ravel(p_prima_simple), width, label='p_prima_simple')

# Dibujar las barras del segundo array
bars2 = ax.bar(x + width/2, np.ravel(p_prima_completo), width, label='p_prima_completo')

ax.set_xlabel('Fila')
ax.set_ylabel('Valores')
ax.set_title('Comparación de Arrays')
ax.set_xticks(x)
ax.set_xticklabels([f'{i}' for i in x])

# Añadir leyenda
ax.legend()

# Mostrar el gráfico
plt.show()

Analisis:

Observamos minimas variaciones dentro de cada modelo. Esto lo asociamos a que las modificaciones en los sectores son menores al 10%  y en 4 de los 40 sectores de la economía. A pesar de que cada delta genera modificaciones mínimas en los sectores en los que afectó el shock, cuando comparamos los resultados de producción entre el modelo simple y completo, notamos que son iguales. Esto podría deberse a que estos no son sectores representativos del comercio entre Venezuela y Costa Rica, por lo que incluir las matrices CC, CV y VC no genera cambios representativos contra el cálculo simple de la producción. La posible causa es la escala de los graficos que muestra valores del orden de $10^6$ hasta aproximadamente $10^7$, entonces una pequeña diferencia resultaria inapreciable en estos gráficos.